llvm/include/llvm/Analysis/ValueLattice.h

```diff
@@ -426,6 +431,16 @@ class ValueLatticeElement {
         return false;
       if (RHS.isUndef())
         return false;
+      // If the constant is a vector of integers, try to treat it as a range.
```

Comment:
Do you know the reason that we weren't allowed to convert a constant into a constant range?

Reply:
For plain integers, constants get converted into constant range on construction. For vectors this doesn't happen as it's a lossy operation. So I think there was just never a need for it before.